hudi-common/src/main/java/org/apache/hudi/common/model/HoodieIndexDefinition.java

```diff
@@ -81,6 +82,10 @@ public String getIndexType() {
     return indexType;
   }
 
+  public boolean isFunctionalIndex() {
+    return indexName.startsWith(PARTITION_NAME_FUNCTIONAL_INDEX_PREFIX);
+  }
```

Comment:
We can use the method in `MetadataPartitionType`

Reply:
I have refactored the method `org.apache.hudi.metadata.MetadataPartitionType#isMetadataPartitionAvailable` to use this API. We need this API because we need to iterate over the functional indexes and then validate each separately.